index.html

```diff
@@ -557,24 +566,66 @@ <h3>Terminology</h3>
 
           <dt><dfn class="export" data-lt="controller(s)|Controllers">controller</dfn></dt>
           <dd>
-An entity that is [=authorized=] to perform an action with a specific resource,
-such as update a [=controller document=] or use a cryptographic key to generate
-a digital signature.
+            <p>
+An entity that is capable of performing an action with a specific resource,
+such as updating a [=controller document=] or generating a [=proof=] using a [=verification method=].
+            </p>
           </dd>
 
           <dt><dfn class="export">controller document</dfn></dt>
           <dd>
-A set of data that specifies one or more relationships between a
-[=controller=] and a set of data, such as a set of public cryptographic keys.
+            <p>
+A document that contains cryptographic material and service
+endpoints for the purposes of verifying proofs from, and interacting
+with, the [=controller=] of an identifier.
+            </p>
           </dd>
 
           <dt><dfn data-lt="subjects">subject</dfn></dt>
           <dd>
-The entity identified by the `id` property in a [=controller document=].
-Anything can be a subject: person, group, organization, physical thing, digital
-thing, logical thing, etc.
+            <p>
+An entity that is referred to by the `id` property in a [=controller document=]
+when it is used as a subject in other contexts, such as authentication or
+attestations, e.g., Verifiable Credentials. Anything can be a subject: 
+person, group, organization, physical thing, digital thing, logical thing, etc. 
+            </p>
+            <p class="note">
+It is expected that the subject referred to by the id of a controller document 
+be consistent over time, so that a set of credentials with a given identifer as 
+subject can be taken to be about the same entity. However, it *is* possible to 
+issue credentials with a subject identifier _without_ the subject even being 
+aware of it.
+
+It is expected that credential issuers will ask the subject for identifiers 
+and that those subjects would demonstrate proof of control before the issuer
+issues the credential. However, there are valid cases where issuers will 
+use identifiers that are not provably under the control of the subject. 
+
+However, even in cases
+where the subject of the ID does prove control, the interpretation of the subject
+remains contextual and potentially ambiguous.
+
+For example, the White House might issue a credential about the president of the United States 
+using did:example:abc as a subject identifier, saying in effect "The President 
+likes fried chicken." Then, the US National Parks Service might issue their own 
+credential about the president, using the same identifier, saying "The President authorizes
+the creation of new National Parks." 
```

Comment:
Importantly, neither agency made a mistake.

References attain meaning through their use. It would be irresponsible of us to support a narrative that assumes an identifier is, and always will be, used exclusively for a particular subject. That is the same fallacy that leads people to believe that information in a database is automatically "true" without a perspective for how facts in the real world are fundamentally changing all the time and, that, in actuality, we cannot know for certain that any given identifier, used by any given party, is in fact referring to the same subject we think it is. Only by knowing the bounds of the actual guarantees of the technology can we effectively apply this technology to real world problems.

Not to get academic, but this phenomenon is called intersubjectivity. It *is* the closest we get to having a shared a cognitive model based on objective reality. What's interesting about intersubjectivity is not that we can never actually be certain what "objective" reality is, but rather that despite the fact that every human being experiences a subjectively different reality, we can achieve effective coordination and reliance on a shared sense of objective truth. See https://direct.mit.edu/books/oa-monograph/5499/Consequences-of-LanguageFrom-Primary-to-Enhanced for a more detailed discussion.

The hard part in this work is getting people to realize that in a decentralized system, there is no one to correct you when you get something wrong. There is no centralized authority who can definitively answer as a proxy for "objective truth". Instead, what we are building is a system where observers/relying parties/verifiers can gather such evidence as satisfies their need for evaluating facts before acting. Some will need stronger evidence. Others less. But none of them should presume for a millisecond that a given identifier is guaranteed to refer to a given subject, ESPECIALLY when it can be used by parties who are neither the controller nor the subject. And that's not even threat modeling key compromise.

Reply:
> References attain meaning through their use.

To a degree. References attain *consistent* meaning through their *description*. This is why we have dictionaries, which, though many treat then as authoritative/controlling, are descriptive/controlled. Dictionaries collect definitions which are based on the (relatively common) use of the words they define.

We use URIs instead of words because words cannot be definitively looked up ("dereferenced") to find out the user's intended meaning. URIs can be dereferenced, which should return useful descriptions/definitions of the intended meaning of each URI, which, in the example here, should have made clear whether `did:example:abc` was intended to identify "whomever occupied the position of POTUS at the time in question" or "President John Q. Smith".

@iherman's analysis matches mine.

Reply:
> This is why we have dictionaries, which, though many treat then as authoritative/controlling, are descriptive/controlled.

Actually, that's not why we have dictionaries, although I understand that is why many people use dictionaries.

We have dictionaries to help you spell or pronounce certain words. The definitions provided are simply examples of meaning commonly associated with a given lexical term. It's one of the reasons that definitions in language dictionaries (instead of technical glossaries) list multiple senses for a given term. Because you *don't* know from the word itself which sense it might mean. Technical dictionaries are different; they are affirmative definitions intended to completely disambiguate within the scope of the technology (generally per specification although reuse is encouraged). Technical dictionaries do not help with open ended vocabularies where anyone can create and use any identifier to make any statement. So, either you're in a rigorous context such as that defined by a spec or you are in an ambiguous context and you need to understand which is which. 

The problem with URLs is that you don't have to control a URL to use it as a subject. You don't have to control a DID to use it as a subject in a VC.

```turtle
<http://whitehouse.gov> rdf:a <http://madeupvocabulary.io/building> .
```

Anyone can make that statement without controlling either of those contexts. In fact, I just did.

As such, the interpretation of the subject of those identifiers is necessarily a disambiguation exercise between you and me (and the white house isn't involved).  Clearly, I meant the physical building that is the White House. However, others might use that same URL to refer to the administrative body known as the White House, aka, the executive branch, e.g., the White House has issued three executive orders this week.

If we somehow restricted statements to require the controller of the subject ID to participate, *then* we might  have a way to make sure that every reference is anchored to a common understanding. 

Now, best practice is to have a resource returned by dereferencing the value of the @context property that describes what the context creator meant by its terms. Of course, that could also be ambiguous and interpreted differently by equally intelligent practitioners of the art. Courts regularly deal with this kind of disambiguation, where the fundamental question is a potential disconnect between the intention of the creators of the law at the time it was created and the facts in the case at hand. It's worth noting that the URL itself of whitehouse.gov does NOT provide any help with disambiguation.

But even *if* the @context is magically unambiguous, we still have the problem that controller identifiers DIDs or URLs are NOT described in any @context. DIDs, by their nature, are context-free identifiers. When we create a DID document, we don't put in a context that someone can use to look up the meaning of the value of an "id" property. Rather, evidence, in the form of VCs, can be gathered to establish a set of claims made using a common identifier, and each and every claim must be independently evaluated as to its appropriateness, including how old it is, who issued it, and the nature of the claim itself, e.g., a "currentHomeAddress" predicate should not be expected to remain true over long timescales. It is the accumulation of these claims that allows us to eventually convince ourselves which are likely about the same subject and which might be about something else.

However, even if you take the position that one of the uses is wrong (and somehow the other one is right), the data modeling and guarantees for VCs, imo, MUST, be exceptionally clear about the fact that an issuer *might* be factually wrong. Because I can make the VC. I can make the statements. And I can be totally wrong about what I say, even just from a typo.

For example, this error has been made by many people, accidentally conflating whitehouse.gov with whitehouse.com.
```turtle
<http://whitehouse.com> rdf:a <http://madeupvocabulary.io/building> .
```
Is it wrong? Well, if you know what whitehouse.com is, it is. But the originator *meant* the white house that is a building in Washington, DC. So they made a valid statement incorrectly.

And yet, the relying party has no programmatic way to disambiguate these kinds of errors. We simply can't presume that it is always correct.

In the same way, reliance on VCs must include an analysis of when the identifiers might mean something slightly different than what you think.

I particular, this touches on the conflation of the credential ID verses the ID of the underlying privilege. Some argue that the credential ID in, for example, a Digital Driver's License should be the Driver's License # (DL #). However, that license # is not referring to that individual credential; it is referring to an underlying account. In California, there is a Document Descriptor (DD) that maps more closely to the "id" field in a credential: it uniquely identifies that particular digital document. Using the DL # as the VC id, instead of a new, unique DD, will lead precisely the kind of error we are talking about.

When we put the DL# as the id, fields like "validUntil" would describe the underlying account identified by the DL#, not this particular digital object. To me, it *SHOULD* be a unique DD, but the imprecise use of the DL# is still a valid identifier and will produce valid and verifiable VCs. So I would argue that you should use the DD, but that is a data modeling choice that isn't even understandable as a choice until you realize that the actual referent of an ID might be ambiguous.

As soon as the DL VC has a different validUntil property than the underlying account, we have no way to know which is which because we built an app that treated it one way and someone else, understandably, built a different app that treated it differently, because of a very subtle nuance about what an identifier refers to.

So, when you see the "id" field, you can't just blindly assume that everyone who is using it is doing so correctly. You can't presume that every VC is a flawless exercise in rigorous attestations. Because people. And errors. They happen. If you presume they don't, you'll have no way to build a system that can be reliable despite these kinds of errors.

Specifically, in the whitehouse.com example, a human operator could simply notice that whitehouse.com doesn't seem to be building and ask for clarification. I know that people want to blindly trust the presumption of correct use of the right identifier at the right time in the right way,  but that will never result in a robust system. And people who think they are getting that robustness for free are going to build systems that fail and blame it on the standard.

It's not the standard that is difficult, it is identity assurance, including asking the question "what did the author/speaker/issuer mean when they used that identifier?" Different contexts will yield different answers, especially when the author meant to make a statement about the website whitehouse.gov, which happens to have the same URL. This is the HttpRange14 problem, which is a direct result of the global context collapse where every identifier must be interpreted as always referring to the same subject. Reality and language simply don't work that way.

Reply:
Sigh. All this is interesting (it really is!) discussion, but I have the feeling that we lose sight of what this discussion is all about. We are discussing an (unusually long, b.t.w.) editorial note in the terminology section which, by its editorial nature, does not add any normative content to our specification. The long discussion has proven that this note leads to potentially long discussions and disagreements; in other words, it does not fulfill its purpose, namely to help to grasp the normative content. My original proposal, namely to remove that note altogether, still stands.

Alternatively, the note can be drastically reduced. The essence of the discussion is the well known fact that for every concept/object/whatever, choosing its "name", in our case its URL (whether it is a DID nor not), must be done with a great care because it can be misunderstood and/or misused. That it is probably a good practice to add additional information to help to disambiguate possible meanings, possibly in the form of VCs, making the URL dereferencable to provide some context, etc. And that is it.

Reply:
> Actually, that's not why we have dictionaries

It's not the *whole* of why we have dictionaries. I should have said "part of why". (Applying correction now.) Different dictionaries (OED, Merriam Webster, etc.) had different creation drivers. Two minutes of web searching should bear me out.

-----

That said, there are a number of typos in your wall of text, above. I urge you to proofread it again, as I am sure I am overlooking some myself, as well as unsure whether I know what the correct text would be in several cases.

Reply:
Two new terms added to the terminology section.